instance-scheduler/main_test.go

```diff
@@ -1,29 +1,17 @@
 package main
 
 import (
-	"fmt"
 	"github.com/aws/aws-lambda-go/events"
-	"net/http"
-	"net/http/httptest"
-	"os"
 	"testing"
 )
 
 func TestHandler(t *testing.T) {
-	os.Setenv("INSTANCE_SCHEDULING_ACTION", "Test")
-
-	t.Run("Successful Request", func(t *testing.T) {
-		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			w.WriteHeader(200)
-			fmt.Fprintf(w, "Testing")
-		}))
-		defer ts.Close()
-
-		DefaultHTTPGetAddress = ts.URL
-
+	t.Run("Testing the lambda function", func(t *testing.T) {
 		_, err := handler(events.APIGatewayProxyRequest{})
 		if err != nil {
-			t.Fatal("Everything should be ok")
```

Comment:
Looking at the documentation below, I think we would want to call Fatal instead of Error, because any follow up test assertions would not be relevant if the API call fails at this stage:

https://pkg.go.dev/testing#B.Error

Error is equivalent to Log followed by Fail.
Fatal is equivalent to Log followed by FailNow.
Fail marks the function as having failed but continues execution.
FailNow marks the function as having failed and stops its execution by calling runtime.Goexit (which then runs all deferred calls in the current goroutine). Execution will continue at the next test or benchmark. FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test. Calling FailNow does not stop those other goroutines.

Reply:
I've used Error, as the plan is to have more tests and we want to run other tests to run and complete and not exit before they have a chance to run. At this stage I can set it to fatal, but it will be changed to error again, once there are more tests added

Reply:
I see where you're coming from. Personally, I would prefer to defer such change until we do add some test assertions as it would be easier to see the behaviour. My guess at this point while we don't have any assertions following the lambda invocation is that, in case of an error at this stage, all of the follow up assertions within the function will always fail. In which case it would be preferable to exit so that it's easier to identify the root of the failure. I might be wrong, but like I said, I would prefer to defer this change until we have some assertions in place.

Reply:
updated